tests/unit/executors/indexers/test_numpyindexer.py

```diff
@@ -160,3 +160,29 @@ def test__get_sorted_top_k(batch_size, num_docs, top_k):
 
         np.testing.assert_equal(idx, expected_idx)
         np.testing.assert_equal(dist, expected_dist)
+
+
+@pytest.mark.parametrize('batch_size, compress_level', [(None, 0), (None, 1), (2, 0), (2, 1)])
+def test_numpy_indexer_empty_data(batch_size, compress_level, test_metas):
+    np.random.seed(500)
+    num_data = 100
+    num_dim = 64
+    num_query = 10
+    vec_idx = np.random.randint(0, high=num_data, size=[num_data])
+    # empty data
+    vec = np.empty((0, 0))
```

Comment:
Are we sure this reflects the problem we faced? Empty shards are empty because they indexed nothing not because they received an empty index query right?

Reply:
Got it

Reply:
it may still be an interesting usecase to test though